### PR TITLE
ci(deploy): #39 Add azure-deploy infrastructure

### LINKED ISSUE #39: Add GH action to run scenario preparation
Add a workflow to run through the workflow on changes to this repo.

### PR DESCRIPTION
**Pull request by AlexAxthelm**
Closes #39 

Add a GH Actions workflow to deploy an Azure Container Instance that runs workflow.scenario.preparation for each of 2022Q4 and 2023Q4 configs.

### CODE REVIEW

File: main.R [hunk 1]
```diff
@@ -47,7 +47,7 @@ config <- config::get(
 system_timestamp <-
   format(
     Sys.time(),
-    format = "%Y%m%d_T%H%M%SZ",
+    format = "%Y%m%dT%H%M%SZ",
```
Comment (cjyetman) [inline main.R:50]: can we leave an underscore or some kind of separator here please? I find the "2023Q420240416T090217Z" format extremely hard to parse visually.

Comment (jdhoffa) [reply main.R]: So, my preferred usage in the past has been separating logically distinct modules with a hyphen, and separating aspects within a module with underscores. 

As an example, for a github PR I opt for:
{issue number}-description_of_pr (e.g. 15-add_unit_tests)

For this use case, that might look like:
2023Q4-20240305_hash
Or
2023Q4-2024_03_05_hash

(Of those two my preference would probably be the former)

Is that helpful?

**Review by cjyetman — APPROVED** [review]
